### PR TITLE
Migrate app-framework

### PR DESCRIPTION
**Pull request by FromLiQg**

### CODE REVIEW

**Comment by FromLiQg** [comment]
I will submit a PR first, but please do not merge

**Review by xwang98 — COMMENTED** [review]
the folder "wamr-sdk" should be moved out?

**Comment by FromLiQg** [comment]
> the folder "wamr-sdk" should be moved out?

yes.
The `wamr-sdk` will be moved to `wamr-app-framework`, but it is expected that `libc-builtin-sysroot` and a copy of `defined-symbol.txt` related to `wamr-app-framework` will be retained in the `wamr`.
@xwang98

File: .github/workflows/nightly_run.yml [hunk 1]
```diff
@@ -449,12 +449,6 @@ jobs:
           cmake ..
           cmake --build . --config Release --parallel 4
           ./hello
-      - name: Build Sample [simple]
-        run: |
-          ./build.sh -p host-interp
-          python3 ./sample_test_run.py $(pwd)/out
-          exit $?
-        working-directory: ./samples/simple
```
Comment (wenyongh) [inline .github/workflows/nightly_run.yml:457]: Had better keep build and run simple in nightly run, but needs to update the script. Maybe we can remove it now and add back later.

File: .github/workflows/release_process.yml [hunk 2]
```diff
@@ -136,38 +136,6 @@ jobs:
...
-
-  release_wamr_sdk_on_macos:
-    needs: [create_tag, create_release]
-    uses: ./.github/workflows/build_wamr_sdk.yml
-    with:
-      config_file: wamr_config_macos_release.cmake
-      runner: macos-latest
-      upload_url: ${{ needs.create_release.outputs.upload_url }}
-      ver_num: ${{ needs.create_tag.outputs.new_ver}}
-      wasi_sdk_url: https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-19/wasi-sdk-19.0-macos.tar.gz
```
Comment (wenyongh) [inline .github/workflows/release_process.yml:169]: Same as above, maybe we need to update the script later to release the wamr-sdk.

File: .gitignore [hunk 3]
```diff
@@ -13,12 +13,9 @@
 
 core/deps/**
 core/shared/mem-alloc/tlsf
-core/app-framework/wgl
 core/iwasm/libraries/lib-wasi-threads/test/*.wasm
 core/iwasm/libraries/lib-socket/test/*.wasm
 
-wamr-sdk/out/
-wamr-sdk/runtime/build_runtime_sdk/
 test-tools/host-tool/bin/
```
Comment (wenyongh) [inline .gitignore:19]: Can also remove host-tool.

File: ci/coding_guidelines_check.py [hunk 4]
```diff
@@ -30,9 +30,7 @@
     "**/samples/wasm-c-api/src/*.*",
     "**/samples/workload/*",
     "**/test-tools/wasi-sdk/*",
-    "**/test-tools/IoT-APP-Store-Demo/*",
```
Comment (wenyongh) [inline ci/coding_guidelines_check.py:33]: Can also remove `assembly-script` in L24.

File: product-mini/README.md [hunk 5]
```diff
@@ -250,9 +250,9 @@ Note:
 WAMR provides some features which can be easily configured by passing options to cmake, please see [WAMR vmcore cmake building configurations](../doc/build_wamr.md#wamr-vmcore-cmake-building-configurations) for details. Currently in VxWorks, interpreter and builtin libc are enabled by default.
 
 ## Zephyr
-
 Please refer to this [README](./platforms/zephyr/simple/README.md) under the Zephyr sample directory for details.
 
+
```
Comment (wenyongh) [inline product-mini/README.md:255]: No need to add this line?

File: wamr-sdk/README.md [hunk 6]
```diff
@@ -1,133 +1,3 @@
...
-# ......
-target_link_libraries (your_target vmlib -lm -ldl -lpthread)
-```
-
-You can refer to to product-mini building for Linux: [`product-mini/platforms/linux/CMakeLists.txt`](../product-mini/platforms/linux/CMakeLists.txt).
-
-> 
+# Notice
+`wamr-sdk` has been migrated to **[wamr-app-framework/wamr-sdk](https://github.com/bytecodealliance/wamr-app-framework/tree/main/wamr-sdk)**.
+Only some necessary files that A will use are retained here
```
Comment (wenyongh) [inline wamr-sdk/README.md:3]: Seems all wamr-sdk files are migrated? Had better remove this line?

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 7]
```diff
@@ -397,13 +395,6 @@ jobs:
           cmake --build . --config Release --parallel 4
           ./hello
 
-      - name: Build Sample [simple]
-        run: |
-          ./build.sh -p host-interp
-          python3 ./sample_test_run.py $(pwd)/out
-          exit $?
-        working-directory: ./samples/simple
```
Comment (wenyongh) [inline .github/workflows/compilation_on_android_ubuntu.yml:405]: Maybe we can update the script later to build simple from wamr-app-framework.

File: .github/workflows/nightly_run.yml (hunk 1 above)
Comment (FromLiQg) [reply .github/workflows/nightly_run.yml:457]: Ok, We add it back later.

File: .github/workflows/release_process.yml (hunk 2 above)
Comment (FromLiQg) [reply .github/workflows/release_process.yml:169]: Ok, We update it back later.

File: .gitignore (hunk 3 above)
Comment (FromLiQg) [reply .gitignore:19]: Ok

File: wamr-sdk/README.md (hunk 6 above)
Comment (FromLiQg) [reply wamr-sdk/README.md:3]: We have retained some files that wasm app will use, like `libc-builtin-sysroot`. So I think we need to keep this line.

File: .github/workflows/compilation_on_android_ubuntu.yml (hunk 7 above)
Comment (FromLiQg) [reply .github/workflows/compilation_on_android_ubuntu.yml:405]: I agree.

**Pull request merged**